src/Projekt/Args.fs

```diff
@@ -1,111 +1,102 @@
 module Projekt.Args
 
 open Projekt.Types
-open System.IO
-open Nessos.UnionArgParser
 
-type private Args =
-    | Template of string
-    | [<AltCommandLine("-fxv")>] FrameworkVersion of string
-    | Organisation of string
-    | Direction of string
-    | Repeat of int
-    | Link of string
-    | Compile of bool
-with
-    interface IArgParserTemplate with
-        member s.Usage = 
-            match s with
-            | Template _ -> "init -- specify the template (library|console) [default: library]"
-            | Direction _ -> "movefile -- specify the direction (down|up)"
-            | Repeat _ -> "movefile -- specify the distance [default: 1]"
-            | FrameworkVersion _ -> "init -- specify the framework version (4.0|4.5|4.5.1) [default: 4.5]"
-            | Organisation _ -> "init -- specify the organisation"
-            | Link _ -> "addfile -- specify an optional Link attribute"
-            | Compile _ -> "addfile -- should the file be compiled or not  [default: true]"
+module Args =
+    open CommandLine
 
-let private templateArg (res : ArgParseResults<Args>) =
-    match res.TryGetResult(<@ Template @>) with
-    | Some (ToLower "console") -> Console
-    | Some (ToLower "library") -> Library
-    | None -> Library
-    | _ -> failwith "invalid template argument specified"
+    [<Verb("init", HelpText = "create a new project")>]
+    type InitOptions = 
+        { [<Value(0, Required = true, MetaName = "project path")>] Path : string 
+          [<Option(Default = "library")>] Template : string
+          [<Option(Default = "4.5")>] FrameworkVersion : string
+          [<Option>] Organization : string option }
+    with 
+        member x.ToOperation =
+            match x.Path with
+            | FullPath p -> 
+                let template = x.Template |> Template.Create |> Some
+                let frmwkVersion = x.FrameworkVersion |> FrameworkVersion.Create |> Some
+                (ProjectInitData.create 
+                    p 
+                    template
+                    frmwkVersion
+                    x.Organization) |> Init
 
-let private frameworkVersionArg (res : ArgParseResults<Args>) =
-    match res.TryGetResult(<@ FrameworkVersion @>) with
-    | Some "4.0" -> V4_0
-    | Some "4.5" -> V4_5 
-    | Some "4.5.1" -> V4_5_1
-    | None -> V4_5
-    | _ -> failwith "invalid framework version argument specified"
+            | _ -> failwith "not given a full path"
 
-let private parseDirection s =
-    match s with
-    | ToLower "up" -> Up
-    | ToLower "down" -> Down
-    | _ -> failwith "invalid direction specified"
+    [<Verb("reference", HelpText = "reference a dll")>]
```

Comment:
Currently this verb is for referencing projects. Referencing dlls is planned: #23 (@kjnilsson you fancy it?)